### PR TITLE
Updating hydro tests

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

File: test/test_device_hydro_generation_constructors.jl [hunk 1]
```diff
@@ -762,19 +767,23 @@ end
 
     for net in networks, mod in models, p in parameters_value
         @testset "$(mod) ED model on $(net) and use_parameters = $(p)" begin
-            devices = Dict{String, DeviceModel}(
-                :Generators => DeviceModel(ThermalStandard, ThermalDispatch),
-                :Loads => DeviceModel(PowerLoad, StaticPowerLoad),
-                :HydroGens => DeviceModel(HydroEnergyReservoir, mod),
-            )
-            template = OperationsProblemTemplate(net, devices, branches, services)
+            template = OperationsProblemTemplate(net)
```
Comment (jd-lara) [inline test/test_device_hydro_generation_constructors.jl:770]: move this template definition to the templates folder. I am trying to reduce the number of templates out there in the tests and their modifications.

File: test/test_device_hydro_generation_constructors.jl [hunk 2]
```diff
@@ -797,19 +806,23 @@ end
 
     for mod in models, p in parameters_value
         @testset "$(mod) ED model on $(net) and use_parameters = $(p)" begin
-            devices = Dict{String, DeviceModel}(
-                :Generators => DeviceModel(ThermalStandard, ThermalDispatch),
-                :Loads => DeviceModel(PowerLoad, StaticPowerLoad),
-                :HydroGens => DeviceModel(HydroEnergyReservoir, mod),
-            )
-            template = OperationsProblemTemplate(net, devices, branches, services)
+            template = OperationsProblemTemplate(net)
```
Comment (jd-lara) [inline test/test_device_hydro_generation_constructors.jl:809]: same here

File: test/test_device_hydro_generation_constructors.jl [hunk 3]
```diff
@@ -819,152 +832,3 @@ end
         end
     end
 end
-
```
Comment (jd-lara) [inline test/test_device_hydro_generation_constructors.jl:822]: Many tests need to enabled and working not removed.

File: test/test_device_hydro_generation_constructors.jl [hunk 4]
```diff
@@ -819,152 +832,3 @@ end
...
-        c_sys5_hyd;
-        use_forecast_data = false,
-    )
-    mock_construct_device!(op_problem, model)
-    moi_tests(op_problem, false, 4, 0, 2, 1, 1, true)
-    psi_checkobjfun_test(op_problem, GAEVF)
-
-end
-
-@testset "Hydro DCPLossLess HydroEnergyReservoir with HydroDispatchReservoirStorage Formulations" begin
```
Comment (jd-lara) [inline test/test_device_hydro_generation_constructors.jl:909]: This is a valid test that needs to be enabled

File: test/test_device_hydro_generation_constructors.jl [hunk 5]
```diff
@@ -819,152 +832,3 @@ end
...
-        c_sys5_hyd;
-        use_forecast_data = false,
-    )
-    mock_construct_device!(op_problem, model)
-    moi_tests(op_problem, false, 4, 0, 2, 1, 1, true)
-    psi_checkobjfun_test(op_problem, GAEVF)
-
-end
-
-@testset "Hydro DCPLossLess HydroEnergyReservoir with HydroCommitmentReservoirlFlow Formulations" begin
```
Comment (jd-lara) [inline test/test_device_hydro_generation_constructors.jl:868]: this is also a valid test that needs enabling

**Review by jd-lara — CHANGES_REQUESTED** [review]
The tests at the bottom need to reviewed and enabled. Please also add the templates to the template file of make minimal modifications. Target minimum code repetition

File: test/test_device_hydro_generation_constructors.jl (hunk 3 above)
Comment (sourabhdalvi) [reply test/test_device_hydro_generation_constructors.jl:822]: Some of them are already present and one of the formulation is no longer valid

File: test/test_device_hydro_generation_constructors.jl (hunk 4 above)
Comment (sourabhdalvi) [reply test/test_device_hydro_generation_constructors.jl:909]: present on line 553

File: test/test_device_hydro_generation_constructors.jl (hunk 5 above)
Comment (sourabhdalvi) [reply test/test_device_hydro_generation_constructors.jl:868]: HydroCommitmentReservoirlFlow was removed in the previous refactor

File: test/test_device_hydro_generation_constructors.jl (hunk 1 above)
Comment (sourabhdalvi) [reply test/test_device_hydro_generation_constructors.jl]: I decided to use the existing template and just add the hydro device to it.

File: test/test_device_hydro_generation_constructors.jl (hunk 5 above)
Comment (jd-lara) [reply test/test_device_hydro_generation_constructors.jl:868]: But we need to enable an equivalent test. Or this is there already?

**Review by jd-lara — APPROVED** [review]

**Pull request merged**